pkg/resource/deploy/source_eval.go

```diff
@@ -1724,6 +1737,29 @@ func (rm *resmon) resolveProvider(
 	return ""
 }
 
+func reportDiagnosticError(sink diag.Sink, urn urn.URN, t tokens.Type, name string, st *status.Status) {
+	message := st.Message()
+	for _, d := range st.Details() {
+		switch d := d.(type) {
+		case *errdetails.BadRequest:
+			message = fmt.Sprintf("%v\n\t\tBadRequest:", message)
```

Comment:
I wonder if we should go more human here if these are going to be printed to the user (e.g. `bad request`, `debug information`, etc.). Or maybe that'll hurt us down the line when it comes to debugging?